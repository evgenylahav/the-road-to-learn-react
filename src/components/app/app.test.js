import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { App } from './app';

import { testNameToKey } from 'jest-snapshot/build/utils';


describe('App', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('checks that only one app is created', () => {
    const element = shallow(
      <App />
    );
    expect(element.find(`[data-test='page']`).length).toBe(1);
  });

  test('has a valid snapshot', () => {
    const component = renderer.create(
      <App />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
