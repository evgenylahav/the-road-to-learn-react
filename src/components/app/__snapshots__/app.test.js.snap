// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App has a valid snapshot 1`] = `
<div
  className="page"
  data-test="page"
>
  <div
    className="interactions"
  >
    <div
      data-test="search-form"
    >
      <form
        onSubmit={[Function]}
      >
        Search
        <input
          onChange={[Function]}
          type="text"
          value="redux"
        />
        <button
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
  </div>
  <div
    className="table"
  >
    <div
      className="table-header"
    >
      <span
        style={
          Object {
            "width": "40%",
          }
        }
      >
        <div>
          <button
            className="button-inline"
            onClick={[Function]}
            type="button"
          >
            Title
          </button>
        </div>
      </span>
      <span
        style={
          Object {
            "width": "30%",
          }
        }
      >
        <div>
          <button
            className="button-inline"
            onClick={[Function]}
            type="button"
          >
            Author
          </button>
        </div>
      </span>
      <span
        style={
          Object {
            "width": "10%",
          }
        }
      >
        <div>
          <button
            className="button-inline"
            onClick={[Function]}
            type="button"
          >
            Comments
          </button>
        </div>
      </span>
      <span
        style={
          Object {
            "width": "10%",
          }
        }
      >
        <div>
          <button
            className="button-inline"
            onClick={[Function]}
            type="button"
          >
            Points
          </button>
        </div>
      </span>
      <span
        style={
          Object {
            "width": "10%",
          }
        }
      >
        Archive
      </span>
    </div>
  </div>
  <div
    className="interactions"
  >
    <div>
      Loading...
    </div>
  </div>
</div>
`;
